src/Bridge/Symfony/Bundle/Resources/config/http_cache_tags.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="api_platform.http_cache.resource_manager" class="ApiPlatform\Core\HttpCache\ResourceManager" public="false" />
+        <service id="api_platform.http_cache.purger.varnish_client" class="GuzzleHttp\Client" abstract="true" public="false" />
+        <service id="api_platform.http_cache.purger.varnish" class="ApiPlatform\Core\HttpCache\VarnishPurger" public="false" />
+
+        <service id="api_platform.http_cache.listener.response.add_tags" class="ApiPlatform\Core\HttpCache\EventListener\AddTagsListener">
+            <argument type="service" id="api_platform.http_cache.resource_manager" />
+            <argument type="service" id="api_platform.iri_converter" />
+
+            <tag name="kernel.event_listener" event="kernel.response" method="onKernelResponse" />
+        </service>
+
+        <!-- The onKernelView method must be executed after the validation (32) but before the write (64) -->
+        <service id="api_platform.http_cache.listener.response.purge" class="ApiPlatform\Core\HttpCache\EventListener\PurgeListener">
+            <argument type="service" id="api_platform.http_cache.resource_manager" />
+            <argument type="service" id="api_platform.http_cache.purger" />
+
+            <tag name="kernel.event_listener" event="kernel.terminate" method="onKernelTerminate" />
```

Comment:
^ That does not actually address the race condition, but only trying to avoid it.